web_export_view/static/src/js/web_export_view.js

```diff
@@ -0,0 +1,104 @@
+odoo.define('web_export_view', function (require) {
+"use strict";
+
+    var core = require('web.core');
+    var Sidebar = require('web.Sidebar');
+    var session = require('web.session');
+
+    var QWeb = core.qweb;
+
+    var _t = core._t;
+
+    Sidebar.include({
+
+        _redraw: function () {
+            var self = this;
+            this._super.apply(this, arguments);
+            if (self.getParent().renderer.viewType == 'list') {
+                self.$el.find('.o_dropdown').last().append(QWeb.render('WebExportTreeViewXls', {widget: self}));
+                self.$el.find('.export_treeview_xls').on('click', self.on_sidebar_export_treeview_xls);
+            }
+        },
+
+        on_sidebar_export_treeview_xls: function () {
+            // Select the first list of the current (form) view
+            // or assume the main view is a list view and use that
+            var self = this,
+                view = this.getParent(),
+                children = view.getChildren();
+            if (children) {
+                children.every(function (child) {
+                    if (child.field && child.field.type == 'one2many') {
+                        view = child.viewmanager.views.list.controller;
+                        return false; // break out of the loop
+                    }
+                    if (child.field && child.field.type == 'many2many') {
+                        view = child.list_view;
+                        return false; // break out of the loop
+                    }
+                    return true;
+                });
+            }
+            var export_columns_keys = [];
+            var export_columns_names = [];
+            var column_index = 0;
+            $.each(view.renderer.columns, function () {
+                if (this.tag == 'field' && (this.attrs.widget === undefined || this.attrs.widget != 'handle')) {
+                    // non-fields like `_group` or buttons
+                    export_columns_keys.push(column_index);
+                    export_columns_names.push(view.$el.find('.o_list_view > thead > tr> th[title]:eq('+column_index+')')[0].textContent);
```

Comment:
I found out this selector only works in debug mode.
I fixed it here https://github.com/acsone/web/commit/d2696b773a02239b334af5328b93c19037461220
I'm not able to push on your repo @JBF91 but you can cherry pick my commit.
This commit also fixes errors when a column is not sortable, this is related to last @CasVissers reported error.